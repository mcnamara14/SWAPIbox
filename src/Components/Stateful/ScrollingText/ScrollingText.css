.scrollingTextContainer {
  background-repeat: repeat !important;
  background: url("./bg.jpg");
  display: block;
  height: 100vh;
  padding-top: 40px; }

#titles {
  bottom: 0;
  font-size: 350%;
  font-weight: bold;
  height: 50em;
  left: 50%;
  margin-left: -9em;
  overflow: hidden;
  position: absolute;
  text-align: justify;
  transform-origin: 50% 100%;
  transform: perspective(300px) rotateX(25deg);
  width: 18em; }

span {
  color: white;
  margin-bottom: 20px; }

.scrollYear {
  color: white;
  font-size: 3rem; }

#titles:after {
  bottom: 0%;
  content: ' ';
  left: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0; }

#titlecontent {
  position: absolute;
  top: 100%;
  animation: scroll 80s linear 4s infinite;
  color: #d39f00; }

@keyframes scroll {
  0% {
    top: 100%; }
  100% {
    top: -170%; } }
