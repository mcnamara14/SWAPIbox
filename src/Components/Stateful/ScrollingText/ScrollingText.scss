$font-stack:    Helvetica, sans-serif;
$primary-color: #333;

.scrollingText {
  width: 20%;
  float: left;
  background: #eee;
  height: 100vh;
}

