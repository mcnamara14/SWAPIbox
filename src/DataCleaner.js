class DataCleaner {
  fetchData = async (filter) => {
    const peopleData = await fetch(`https://swapi.co/api/${filter}`)
    const response = await peopleData.json()
    const rawData = await response.results

    switch(filter) {
      case 'people':
          return this.cleanPeopleData(rawData)
          break;
      default:
          null
    }

  };



  fetchPeopleData = (people) => {
    const unresolvedPeopleData = people.map(async (person) => {
      const homeworldResponse = await fetch(person.homeworld);
      const parsedHomeworld = await homeworldResponse.json();
      const speciesResponse = await fetch(person.species);
      const parsedSpecies = await speciesResponse.json();

      return {name: person.name, homeworld: parsedHomeworld.name, 
              species: parsedSpecies.name, population: parsedHomeworld.population}
    });

    return Promise.all(unresolvedPeopleData)
  }

  cleanPeopleData = async (rawData) => {
    const cleanPeopleData = await this.fetchPeopleData(rawData)
    
    return cleanPeopleData
  }

}

export default DataCleaner;