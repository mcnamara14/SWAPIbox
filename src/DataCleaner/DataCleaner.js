class DataCleaner {
  fetchData = async (filter) => {
    const fetchedData = await fetch(`https://swapi.co/api/${filter}`);
    const response = await fetchedData.json();
    const rawData = response.results;

    switch (filter) {
      case 'people':
        return this.cleanPeopleData(rawData);
      case 'planets':
        return this.cleanPlanetData(rawData);
      case 'vehicles':
        return this.cleanVehicleData(rawData);
    }
  };

  cleanPeopleData = async (people) => {
    const unresolvedPeopleData = people.map(async (person, index) => {
      const name = person.name;
      const homeworldData = await this.cleanHomeworldData(person.homeworld);
      const homeworldName = homeworldData.name;
      const homeworldPopulation = homeworldData.population;
      const species = await this.cleanSpeciesData(person.species);
      
      return {name, homeworld: homeworldName, 
        species, population: homeworldPopulation, id: `people${index}`};
    });

    return await Promise.all(unresolvedPeopleData);
  };

  cleanHomeworldData = async (homeworld) => {
    const homeworldResponse = await fetch(homeworld);
    const homeworldData = await homeworldResponse.json();
    const name = homeworldData.name;
    const population = homeworldData.population;

    return { name, population };
  }

  cleanSpeciesData = async (species) => {
    const speciesResponse = await fetch(species);
    const speciesData = await speciesResponse.json();
    const speciesName = speciesData.name;

    return speciesName;
  }

  cleanPlanetData = async (planets) => {
    const unresolvedPlanetData = planets.map(async (planet, index) => {
      const residents = await this.cleanResidentsData(planet);
 
      return {name: planet.name, terrain: planet.terrain, 
        population: planet.population, climate: planet.climate, 
        residents,  id: `planets${index}`};
    });

    return await Promise.all(unresolvedPlanetData);
  };

  cleanResidentsData = async (planet) => {
    const unresolvedResidents = planet.residents.map(async (resident) => {
      const residentsResponse = await fetch(resident);
      const parsedResident = await residentsResponse.json();
      const residentName = parsedResident.name;

      return residentName;
    });

    return await Promise.all(unresolvedResidents);
  };

  cleanVehicleData = async (vehicles) => {
    const unresolvedVehiclesData = vehicles.map(async (vehicle, index) => {
      return {name: vehicle.name, model: vehicle.model, 
        class: vehicle.vehicle_class, passengers: vehicle.passengers, 
        id: `vehicles${index}`};
    });

    return await Promise.all(unresolvedVehiclesData);
  }
 
}

export default DataCleaner;