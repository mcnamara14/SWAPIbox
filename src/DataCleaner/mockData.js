export const mockCleanedVehicleData = {
    class: "sail barge",
    id: "vehicles0",
    model: "Modified Luxury Sail Barge",
    name: "Sail barge",
    passengers: "500"
}

export const mockCleanedPeopleData = {
  homeworld: "Tatooine",
  id: "people0",
  name: "Luke Skywalker",
  population: "200000",
  species: "Human"
}

export const mockCleanedPlanetData = {
  climate: "temperate",
  id: "planets0",
  name: "Alderaan",
  residents: "Alderaan",
  population: "2000000000",
  terrain: "grasslands, mountains"
}

export const mockDirtyPlanetData = {
  "count": 61,
  "next": "https://swapi.co/api/planets/?page=2",
  "previous": null,
  "results": [
      {
        "name": "Alderaan",
        "rotation_period": "24",
        "orbital_period": "364",
        "diameter": "12500",
        "climate": "temperate",
        "gravity": "1 standard",
        "terrain": "grasslands, mountains",
        "surface_water": "40",
        "population": "2000000000",
        "residents": [
            "https://swapi.co/api/people/5/",
            "https://swapi.co/api/people/68/",
            "https://swapi.co/api/people/81/"
        ],
        "films": [
            "https://swapi.co/api/films/6/",
            "https://swapi.co/api/films/1/"
        ],
        "created": "2014-12-10T11:35:48.479000Z",
        "edited": "2014-12-20T20:58:18.420000Z",
        "url": "https://swapi.co/api/planets/2/"
      }
    ]
  }

export const mockDirtyPeopleData = {
  "count": 87,
  "next": "https://swapi.co/api/people/?page=2",
  "previous": null,
  "results": [
      {
          "name": "Luke Skywalker",
          "height": "172",
          "mass": "77",
          "hair_color": "blond",
          "skin_color": "fair",
          "eye_color": "blue",
          "birth_year": "19BBY",
          "gender": "male",
          "homeworld": "https://swapi.co/api/planets/1/",
          "films": [
              "https://swapi.co/api/films/2/",
              "https://swapi.co/api/films/6/",
              "https://swapi.co/api/films/3/",
              "https://swapi.co/api/films/1/",
              "https://swapi.co/api/films/7/"
          ],
          "species": [
              "https://swapi.co/api/species/1/"
          ],
          "vehicles": [
              "https://swapi.co/api/vehicles/14/",
              "https://swapi.co/api/vehicles/30/"
          ],
          "starships": [
              "https://swapi.co/api/starships/12/",
              "https://swapi.co/api/starships/22/"
          ],
          "created": "2014-12-09T13:50:51.644000Z",
          "edited": "2014-12-20T21:17:56.891000Z",
          "url": "https://swapi.co/api/people/1/"
      }
  ]
}

export const mockDirtyVehicleData = {
  "count": 39,
  "next": "https://swapi.co/api/vehicles/?page=2",
  "previous": null,
  "results": [
      {
          "name": "Sail barge",
          "model": "Modified Luxury Sail Barge",
          "manufacturer": "Ubrikkian Industries Custom Vehicle Division",
          "cost_in_credits": "285000",
          "length": "30",
          "max_atmosphering_speed": "100",
          "crew": "26",
          "passengers": "500",
          "cargo_capacity": "2000000",
          "consumables": "Live food tanks",
          "vehicle_class": "sail barge",
          "pilots": [],
          "films": [
              "https://swapi.co/api/films/3/"
          ],
          "created": "2014-12-18T10:44:14.217000Z",
          "edited": "2014-12-22T18:21:15.807906Z",
          "url": "https://swapi.co/api/vehicles/24/"
      }
  ]
}

export const mockDirtyHomeworldData = {
  "name": "Tatooine",
  "rotation_period": "23",
  "orbital_period": "304",
  "diameter": "10465",
  "climate": "arid",
  "gravity": "1 standard",
  "terrain": "desert",
  "surface_water": "1",
  "population": "200000",
  "residents": [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/2/",
      "https://swapi.co/api/people/4/",
      "https://swapi.co/api/people/6/",
      "https://swapi.co/api/people/7/",
      "https://swapi.co/api/people/8/",
      "https://swapi.co/api/people/9/",
      "https://swapi.co/api/people/11/",
      "https://swapi.co/api/people/43/",
      "https://swapi.co/api/people/62/"
  ],
  "films": [
      "https://swapi.co/api/films/5/",
      "https://swapi.co/api/films/4/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/"
  ],
  "created": "2014-12-09T13:50:49.641000Z",
  "edited": "2014-12-21T20:48:04.175778Z",
  "url": "https://swapi.co/api/planets/1/"
}

export const mockDirtySpeciesData = {
  "name": "Droid",
  "classification": "artificial",
  "designation": "sentient",
  "average_height": "n/a",
  "skin_colors": "n/a",
  "hair_colors": "n/a",
  "eye_colors": "n/a",
  "average_lifespan": "indefinite",
  "homeworld": null,
  "language": "n/a",
  "people": [
      "https://swapi.co/api/people/2/",
      "https://swapi.co/api/people/3/",
      "https://swapi.co/api/people/8/",
      "https://swapi.co/api/people/23/",
      "https://swapi.co/api/people/87/"
  ],
  "films": [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/7/",
      "https://swapi.co/api/films/5/",
      "https://swapi.co/api/films/4/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/"
  ],
  "created": "2014-12-10T15:16:16.259000Z",
  "edited": "2015-04-17T06:59:43.869528Z",
  "url": "https://swapi.co/api/species/2/"
}

export const mockDirtyFilmsData = [
    {
        "title": "A New Hope",
        "episode_id": 4,
        "opening_crawl": "It is a period of civil war to the galaxy....",
        "director": "George Lucas",
        "producer": "Gary Kurtz, Rick McCallum",
        "release_date": "1977-05-25",
        "characters": [
            "https://swapi.co/api/people/1/",
            "https://swapi.co/api/people/2/"
        ],
        "planets": [
            "https://swapi.co/api/planets/2/"
        ],
        "starships": [
            "https://swapi.co/api/starships/2/",
            "https://swapi.co/api/starships/3/",
            "https://swapi.co/api/starships/5/"
        ],
        "vehicles": [
            "https://swapi.co/api/vehicles/4/"
        ],
        "species": [
            "https://swapi.co/api/species/5/",
            "https://swapi.co/api/species/3/"
        ],
        "created": "2014-12-10T14:23:31.880000Z",
        "edited": "2015-04-11T09:46:52.774897Z",
        "url": "https://swapi.co/api/films/1/"
    }
]

export const mockDirtyResidentData = {

}