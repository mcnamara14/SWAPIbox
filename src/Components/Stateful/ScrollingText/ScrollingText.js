import React, { Component } from 'react';
import './ScrollingText.css';
import themeSong from './imperial-march.mp3';

class ScrollingText extends Component {
  constructor() {
    super();
    
    this.state = {
      crawl: '', 
      title: '',
      date: ''
    }
  }
    
  componentDidMount = () => {
    fetch('https://swapi.co/api/films')
      .then(response => response.json())
      .then(response => this.returnRandomCrawl(response.results))
  }

  returnRandomCrawl = (films) => {
    const filmsLength = films.length;
    const randomFilm = Math.floor(Math.random() * filmsLength) + 0;
    const crawl = films[randomFilm].opening_crawl;
    const title = films[randomFilm].title;
    const date = films[randomFilm].release_date;

    this.setState({ crawl, title, date })
  }

  render() {
    return (
      <section className="scrollingTextContainer">
      <div id="titles">
        <div id="titlecontent">
        <audio src={themeSong} autoPlay loop></audio>
        <p>{this.state.crawl}</p>
        <p><span>{this.state.title}</span></p>
        <p className="scrollYear"><span>{this.state.date}</span></p>
    
      </div>
    </div>
    </section>

    );
  };
}

export default ScrollingText;
