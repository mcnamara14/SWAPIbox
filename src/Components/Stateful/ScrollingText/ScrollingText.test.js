import React from 'react';
import ReactDOM from 'react-dom';
import ScrollingText from './ScrollingText';
import { shallow } from 'enzyme';
import { mockDirtyFilmsData } from '../../../DataCleaner/mockData';

describe('ScrollingText', () => {
  beforeEach(() => {
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
        json: () => Promise.resolve(
          mockDirtyFilmsData
      )
    }))
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<ScrollingText />, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  // describe('componentDidMount', async () => {
  //   const wrapper = shallow(<ScrollingText />);
  //   it('should call returnRandomCrawl after fetch', () => {
  //     wrapper.instance().returnRandomCrawl = jest.fn();
  //     window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
  //       status: 200,
  //       json: () => Promise.resolve(
  //         wrapper.instance().returnRandomCrawl
  //       )
  //     })
  //   )
  //     expect(wrapper.instance().returnRandomCrawl).toHaveBeenCalled();
  //   })
  // })

  describe('returnRandomCrawl', () => {
    it('should set state to random crawl data', () => {
      const wrapper = shallow(<ScrollingText />, { disableLifecycleMethods: true });
      const films = mockDirtyFilmsData;
      const expectedCrawl = films[0].opening_crawl;
      const expectedTitle = films[0].title;
      const expectedDate = films[0].release_date;

      wrapper.instance().returnRandomCrawl(films);

      expect(wrapper.state().crawl).toEqual(expectedCrawl)
      expect(wrapper.state().title).toEqual(expectedTitle)
      expect(wrapper.state().date).toEqual(expectedDate)
    })
  })

  
})