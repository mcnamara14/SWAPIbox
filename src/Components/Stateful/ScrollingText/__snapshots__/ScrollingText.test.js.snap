// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollingText />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="titles"
>
        <div
                id="titlecontent"
        >
                <p />
                <p>
                        <span />
                </p>
                <p
                        className="scrollYear"
                >
                        <span />
                </p>
                <audio
                        autoPlay={true}
                        loop={true}
                        src="imperial-march.mp3"
                />
        </div>
</div>,
      "className": "scrollingTextContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="titlecontent"
>
          <p />
          <p>
                    <span />
          </p>
          <p
                    className="scrollYear"
          >
                    <span />
          </p>
          <audio
                    autoPlay={true}
                    loop={true}
                    src="imperial-march.mp3"
          />
</div>,
        "id": "titles",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            <p>
              <span />
</p>,
            <p
              className="scrollYear"
>
              <span />
</p>,
            <audio
              autoPlay={true}
              loop={true}
              src="imperial-march.mp3"
/>,
          ],
          "id": "titlecontent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span />,
              "className": "scrollYear",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoPlay": true,
              "loop": true,
              "src": "imperial-march.mp3",
            },
            "ref": null,
            "rendered": null,
            "type": "audio",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="titles"
>
          <div
                    id="titlecontent"
          >
                    <p />
                    <p>
                              <span />
                    </p>
                    <p
                              className="scrollYear"
                    >
                              <span />
                    </p>
                    <audio
                              autoPlay={true}
                              loop={true}
                              src="imperial-march.mp3"
                    />
          </div>
</div>,
        "className": "scrollingTextContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="titlecontent"
>
            <p />
            <p>
                        <span />
            </p>
            <p
                        className="scrollYear"
            >
                        <span />
            </p>
            <audio
                        autoPlay={true}
                        loop={true}
                        src="imperial-march.mp3"
            />
</div>,
          "id": "titles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p />,
              <p>
                <span />
</p>,
              <p
                className="scrollYear"
>
                <span />
</p>,
              <audio
                autoPlay={true}
                loop={true}
                src="imperial-march.mp3"
/>,
            ],
            "id": "titlecontent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
                "className": "scrollYear",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoPlay": true,
                "loop": true,
                "src": "imperial-march.mp3",
              },
              "ref": null,
              "rendered": null,
              "type": "audio",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
